elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideStandaloneSettings.java

```diff
@@ -400,35 +409,41 @@ default Cache getQueryCache() {
      * @param entityManagerFactory EntityManagerFactory object.
      * @return EntityDictionary object initialized.
      */
-    default DataStore getDataStore(MetaDataStore metaDataStore, AggregationDataStore aggregationDataStore,
-            EntityManagerFactory entityManagerFactory) {
+    default DataStore getDataStore(Optional<MetaDataStore> metaDataStore,
+            Optional<AggregationDataStore> aggregationDataStore, EntityManagerFactory entityManagerFactory) {
         DataStore jpaDataStore = new JpaDataStore(
                 () -> { return entityManagerFactory.createEntityManager(); },
                 (em) -> { return new NonJtaTransaction(em, TXCANCEL); });
 
-        DataStore dataStore = new MultiplexManager(jpaDataStore, metaDataStore, aggregationDataStore);
-
-        return dataStore;
+        return (enableAggregationDataStore())
+                        ? new MultiplexManager(jpaDataStore, metaDataStore.get(), aggregationDataStore.get())
```

Comment:
If the enableAggregationDataStore is enabled, but the aggregation store or metadata store is not provided, we need a good way to alert the user to their configuration error.

Instead of using so many optionals, I'm wondering if we should provide different getDataStore methods with different required parameters (that we use Precondition to check are non-null).  Elide can call the right one depending on what is enabled/disabled.

Reply:
overloaded getDataStore method to avoid Optionals

Reply:
Added null check for metadata store and aggregation store. Without these checks also application was failing to start by giving NPE during queryEngine.init and populateEntityDictionary methods.